modules/tinymce/src/plugins/table/test/ts/browser/command/MergeCellCommandTest.ts

```diff
@@ -35,68 +31,153 @@ UnitTest.asynctest('browser.tinymce.plugins.table.command.MergeCellCommandTest',
           '</table>'
-        ),
-        after: (
-          '<table>' +
+      ),
+      after: (
+        '<table>' +
           '<tbody>' +
           '<tr><td colspan="2" rowspan="2">' +
           'a1<br />b1<br />a2<br />b2' +
           '</td></tr><tr></tr>' +
           '</tbody>' +
           '</table>'
-        ),
-        expectedEvents: [ defaultEvent ],
-      },
-      {
-        message: 'Should merge cells in two cols/rows into one cell with colspan',
-        before: (
-          '<table>' +
+      ),
+      expectedEvents: [ defaultEvent ],
+    });
+  });
+
+  it('TBA: Should merge cells in two cols/rows into one cell with colspan', () => {
+    const editor = hook.editor();
+    testMerge(editor, {
+      before: (
+        '<table>' +
           '<tbody>' +
           '<tr><td data-mce-selected="1" data-mce-first-selected="1">a1</td><td data-mce-selected="1">b1</td></tr>' +
           '<tr><td data-mce-selected="1">a2</td><td data-mce-selected="1" data-mce-last-selected="1">b2</td></tr>' +
           '<tr><td>a3</td><td>b3</td></tr>' +
           '</tbody>' +
           '</table>'
-        ),
-        after: (
-          '<table>' +
+      ),
+      after: (
+        '<table>' +
           '<tbody>' +
           '<tr><td colspan="2" rowspan="2">a1<br />b1<br />a2<br />b2</td></tr>' +
           '<tr></tr>' +
           '<tr><td>a3</td><td>b3</td></tr>' +
           '</tbody>' +
+        '</table>'
+      ),
+      expectedEvents: [ defaultEvent ],
+    });
+  });
+
+  it('TBA: Should merge b3+c3 but not reduce a2a3', () => {
+    const editor = hook.editor();
+    testMerge(editor, {
+      before: (
+        '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td>a1</td>' +
+                '<td>b1</td>' +
+                '<td>c1</td>' +
+              '</tr>' +
+              '<tr>' +
+                '<td rowspan="2">a2a3</td>' +
+                '<td>b2</td>' +
+                '<td>c2</td>' +
+              '</tr>' +
+              '<tr>' +
+                '<td data-mce-selected="1" data-mce-first-selected="1">b3</td>' +
+                '<td data-mce-selected="1" data-mce-last-selected="1">c3</td>' +
+              '</tr>' +
+            '</tbody>' +
           '</table>'
-        ),
-        expectedEvents: [ defaultEvent ],
-      },
-      /*
+      ),
+      after: (
+        '<table>' +
+          '<tbody>' +
+            '<tr>' +
+              '<td>a1</td>' +
+              '<td>b1</td>' +
+            '<td>c1</td>' +
+            '</tr>' +
+            '<tr>' +
+              '<td rowspan="2">a2a3</td>' +
+              '<td>b2</td>' +
+              '<td>c2</td>' +
+            '</tr>' +
+            '<tr>' +
+              '<td colspan="2">b3<br />c3</td>' +
+            '</tr>' +
+          '</tbody>' +
+        '</table>'
+      ),
+      expectedEvents: [ defaultEvent ],
+    });
+  });
+
+  it('TINY-6901: Will apply correct size to cell with colspan after cell merge', () => {
```

Comment:
This will almost definitely fail on other browsers, as you're going to have issues with precision for floating-point numbers.  I'd suggest using a fixed-width table instead, as it's much less likely to have issues.

Reply:
Ignore I missed it had custom logic sorry.